python/lsst/pipe/base/__init__.py

```diff
@@ -1,4 +1,5 @@
 from . import automatic_connection_constants, connectionTypes, pipeline_graph, pipelineIR
+from ._check_qg_outputs import *
```

Comment:
Now that we've settled on names for the classes, we should rename the module to match; how about `execution_reports`?

I've left out the underscore because I also think we should remove it from this `__init__.py`, and instead expect users to do
```
from lsst.pipe.base.execution_reports import QuantumGraphExecutionReport
```
(etc.)

That's a bit more verbose, but since the vast majority of `lsst.pipe.base` imports aren't going to involve this module, it's best to not have all of its import-time logic executed all the time.